### DISCUSSION TITLE
How to catch ClientResponseError?

### DISCUSSION THREAD

**Discussion by dsm-72** [pocketbase/pocketbase#1617] [Q&A]
Just want to try and get a record and if not found, try again with a `~`

**Comment by ganigeorgiev** [comment]
You should be able to catch it like any other JS error. You can chain a `catch()` to the promise or wrap with `try/catch` if you are using the async/await syntax:

```js
import PocketBase, { ClientResponseError } from "pocketbase";

...

try {
    await pb.collection("example").getOne("RECORD_ID");
} catch (err) {
    console.log(err instanceof ClientResponseError);
}
```

Note that all client errors are returned and normalized as `ClientResponseError`. You can find more information in https://github.com/pocketbase/js-sdk#error-handling.

---

btw. depending on your use case, you can also try combining your conditions with the `||` operator instead of sending 2 requests:

```js
const record = await pb.collection("example").getFirstListItem(`id = "${idVar}" || name ~ "${nameVar}"`);
```

**Comment by dsm-72** [comment]
Thanks @ganigeorgiev. I didn't know where to import it from.
Yes, I could use the or condition, but in this case I have a collection called `slugs` which is basically the valid routes for my application and am searching first for an exact match, then for a close match (and redirecting), before finally throwing a 404. Sure I could combine the exact match and close match in a single statement `(slug = "${slug}") || (slug ~ ${slug})` but I am not sure how nicely that will play with the way I have redirects currently set up. I should probably refractor that...
